frameworks/Dart/aqueduct/server.dart

```diff
@@ -0,0 +1,228 @@
+import 'dart:async';
+import 'dart:io';
+import 'dart:math';
+
+import 'package:aqueduct/aqueduct.dart';
+import 'package:mustache/mustache.dart' as mustache;
+
+export 'dart:async';
+export 'dart:io';
+export 'package:aqueduct/aqueduct.dart';
+
+final _stripped_text = new ContentType("text", "plain");
+final _stripped_json = new ContentType("application", "json");
+
+final _random = new Random();
+
+// World table size
+const int _world_table_size = 10000;
+// Fortune table size used only for generation of data
+const int _FORTUNE_TABLE_SIZE = 100;
+
+const int _min_query_count = 1;
+const int _max_query_count = 500;
+
+Future main() async {
+  try {
+    var app = new Application<DartAqueductBenchmarkSink>();
+    var config = new ApplicationConfiguration()
+      ..port = 8080
+      ..configurationFilePath = "config.yaml";
+
+    app.configuration = config;
+    await app.start(numberOfInstances: 3, consoleLogging: false);
+  } catch (e, st) {
+    await writeError("$e\n $st");
+  }
+}
+
+Future writeError(String error) async {
+  print("$error");
+}
+
+class Fortune extends ManagedObject<_Fortune>
+    implements _Fortune, Comparable<Fortune> {
+  compareTo(Fortune other) => message.compareTo(other.message);
+}
+
+class _Fortune {
+  static String tableName() => "fortune";
+
+  @ManagedColumnAttributes(primaryKey: true)
+  int id;
+
+  String message;
+}
+
+class World extends ManagedObject<_World> implements _World {}
+
+class _World {
+  static String tableName() => "world";
+
+  @ManagedColumnAttributes(primaryKey: true)
+  int id;
+
+  int randomNumber;
+}
+
+/// This type initializes an application.
+///
+/// Override methods in this class to set up routes and initialize resources like
+/// database connections. See http://aqueduct.io/docs/http/request_sink.
+class DartAqueductBenchmarkSink extends RequestSink {
+  mustache.Template fortunesTemplate;
+
+  /// Resource initialization code goes here.
+  ///
+  /// Resources like [AuthServer] and [PostgreSQLPersistentStore] should be instantiated
+  /// in this constructor. Configuration behavior - like [HTTPCodecRepository.add] - should be
+  /// configured in this constructor.
+  ///
+  /// The [appConfig] contains configuration data from `aqueduct serve`, e.g.
+  /// the port the application is running on and the path to a configuration file.
+  DartAqueductBenchmarkSink(ApplicationConfiguration appConfig)
+      : super(appConfig) {
+    // Use logging for debuging only
+//    logger.onRecord.listen(
+//        (rec) => print("$rec ${rec.error ?? ""} ${rec.stackTrace ?? ""}"));
+
+    var options =
+        new DartAqueductBenchmarkConfiguration(appConfig.configurationFilePath);
+    ManagedContext.defaultContext = contextWithConnectionInfo(options.database);
+
+    Future.wait([
```

Comment:
You can override `willOpen` in this class, which is an async initialization method, allowing you to await on this Future.

Reply:
(or you can use readAsStringSync)

Reply:
Thanks